packages/Search/src/details/DTK_DetailsUtils.hpp

```diff
@@ -217,6 +217,44 @@ accumulate( ViewType const &v, typename ViewType::non_const_value_type init )
     return init;
 }
 
+// FIXME shameless forward declaration
+template <typename View>
+typename View::non_const_type clone( View &v );
+
+/** \brief Computes the adjacent difference.
+ *
+ *  \param[in] src Input view
+ *  \param[out] dst Output view; may be equal to \p src
+ *
+ *  Assigns to every element in the \p dst view the difference between its
+ *  corresponding element and the one preceding it in the \p src view.
+ */
+template <typename ViewType>
+void adjacentDifference( ViewType const &src, ViewType const &dst )
```

Comment:
Allow for different types for `src` and `dst` (i.e. `ViewType1` and `ViewType2`.
Consider `adjacentDifference( v )` for in-place.